src/evergreen/api.py

```diff
@@ -799,6 +801,71 @@ def manifest_for_task(self, task_id: str) -> Manifest:
         url = self._create_url(f"/tasks/{task_id}/manifest")
         return Manifest(self._call_api(url).json(), self)  # type: ignore[arg-type]
 
+    def get_task_annotation(
+        self,
+        task_id: str,
+        execution: Optional[int] = None,
+        fetch_all_executions: Optional[bool] = None,
+    ) -> List[TaskAnnotation]:
+        """
+        Get the task annotations for the given task.
+
+        :param task_id: Id of task to query.
+        :param execution: Execution number of task to query (defaults to 0).
+        :param fetch_all_executions: Get annotations for all executions of this task.
+        :return: The task annotations for the given task, if any exists.
+        """
+        url = self._create_url(f"/tasks/{task_id}/annotations")
+        params: Dict[str, Any] = {}
+        if execution:
+            params["execution"] = execution
+        if fetch_all_executions:
+            params["fetch_all_executions"] = fetch_all_executions
```

Comment:
If fetch_all_execution is true and execution is defined, you will bump into [this error](https://github.com/evergreen-ci/evergreen/blob/master/rest/route/annotations.go#L164). You might want to do some checking on your end to prevent that.

Reply:
I've added a check for that.